xbmc/utils/Observer.cpp

```diff
@@ -24,72 +24,20 @@
 
 #include <algorithm>
 
-Observer::~Observer(void)
-{
-  StopObserving();
-}
-
-void Observer::StopObserving(void)
-{
-  CSingleLock lock(m_obsCritSection);
-  for (auto& observable : m_observables)
-    observable->UnregisterObserver(this);
-}
-
-bool Observer::IsObserving(const Observable &obs) const
-{
-  CSingleLock lock(m_obsCritSection);
-  return find(m_observables.begin(), m_observables.end(), &obs) != m_observables.end();
-}
-
-void Observer::RegisterObservable(Observable *obs)
-{
-  CSingleLock lock(m_obsCritSection);
-  if (!IsObserving(*obs))
-    m_observables.push_back(obs);
-}
-
-void Observer::UnregisterObservable(Observable *obs)
-{
-  CSingleLock lock(m_obsCritSection);
-  std::vector<Observable *>::iterator it = find(m_observables.begin(), m_observables.end(), obs);
-  if (it != m_observables.end())
-    m_observables.erase(it);
-}
-
-Observable::Observable() :
-    m_bObservableChanged(false)
-{
-}
-
-Observable::~Observable()
-{
-  StopObserver();
-}
-
 Observable &Observable::operator=(const Observable &observable)
 {
   CSingleLock lock(m_obsCritSection);
 
-  m_bObservableChanged = observable.m_bObservableChanged;
-  m_observers.clear();
-  for (unsigned int iObsPtr = 0; iObsPtr < observable.m_observers.size(); iObsPtr++)
-    m_observers.push_back(observable.m_observers.at(iObsPtr));
+  m_bObservableChanged = static_cast<bool>(observable.m_bObservableChanged);
+  m_observers = observable.m_observers;
 
   return *this;
 }
 
-void Observable::StopObserver(void)
-{
-  CSingleLock lock(m_obsCritSection);
-  for (auto& observer : m_observers)
-    observer->UnregisterObservable(this);
-}
-
 bool Observable::IsObserving(const Observer &obs) const
 {
   CSingleLock lock(m_obsCritSection);
-  return find(m_observers.begin(), m_observers.end(), &obs) != m_observers.end();
+  return std::find(m_observers.begin(), m_observers.end(), &obs) != m_observers.end();
 }
```

Comment:
Doesn't the act that this used to compile at all mean we have a `using namespace std;` in a header somewhere?